tests/pytests/integration/states/test_file.py

```diff
@@ -480,6 +480,55 @@ def test_issue_60203(
         )
 
 
+def test_clean_with_wildcard_requisites(
+    salt_master,
+    salt_call_cli,
+    pillar_tree,
+    tmp_path,
+    salt_minion,
+):
+    create_file = tmp_path / "foo"
+
+    sls_name = "file-directory-clean-wildcard"
+    sls_contents = """
+    {tmp_path}:
+      file.directory:
+        - clean: true
+        - require:
+          - file: {tmp_path}/*
+
+    {create_file}:
+      file.managed:
+        - contents: foo bar baz
+    """.format(
+            tmp_path=tmp_path,
+            create_file=create_file,
+    )
+
+    sls_tempfile = salt_master.state_tree.base.temp_file(
+        "{}.sls".format(sls_name), sls_contents
+    )
+    # A file that we expect to be cleaned up
+    clean_me = pytest.helpers.temp_file("bar", "bar", tmp_path)
+
+    with sls_tempfile, clean_me as clean_path:
+        ret = salt_call_cli.run("state.apply", sls_name)
+        assert ret.exitcode == 0
+        assert ret.json
+        states = iter(ret.json.values())
+
+        # Ensure the file was created
+        state_run = next(states)
+        assert state_run["result"] is True
+        assert create_file.is_file()
+
+        # Check to make sure the untracked file was cleaned up
+        state_run = next(states)
+        assert not clean_path.exists()
+        assert "removed" in state_run["changes"]
+        assert str(clean_path) in state_run["changes"]["removed"]
```

Comment:
Windon't
```
[2022-09-19T20:34:44.657Z]        >           assert str(clean_path) in state_run["changes"]["removed"]
[2022-09-19T20:34:44.657Z]        E           AssertionError: assert 'C:\\Users\\Administrator\\AppData\\Local\\Temp\\pytest-of-Administrator\\pytest-0\\test_clean_with_wildcard_requi0\\bar' in ['c:\\users\\administrator\\appdata\\local\\temp\\pytest-of-administrator\\pytest-0\\test_clean_with_wildcard_requi0\\bar']
[2022-09-19T20:34:44.657Z]        E            +  where 'C:\\Users\\Administrator\\AppData\\Local\\Temp\\pytest-of-Administrator\\pytest-0\\test_clean_with_wildcard_requi0\\bar' = str(WindowsPath('C:/Users/Administrator/AppData/Local/Temp/pytest-of-Administrator/pytest-0/test_clean_with_wildcard_requi0/bar'))
```

Reply:
It seems the state_run path is all lowercase, so maybe converting the LHS to lower will make it pass?